crates/universal-sierra-compiler-api/src/utils.rs

```diff
@@ -0,0 +1,30 @@
+use anyhow::Result;
+use camino::Utf8PathBuf;
+use indicatif::{ProgressBar, ProgressStyle};
+use std::env;
+use std::time::Duration;
+
+pub fn get_spinner_message(sierra_file_path: &Utf8PathBuf) -> Result<ProgressBar> {
+    let spinner = ProgressBar::new_spinner();
+    spinner.set_style(ProgressStyle::with_template("\n{spinner} {msg}\n")?);
+    spinner.enable_steady_tick(Duration::from_millis(100));
+
+    // Skip printing path when compiling unsaved sierra
+    // which occurs during test execution for some cheatcodes e.g. `replace_bytecode`
+    let message = if is_temp_file(sierra_file_path)? {
+        "Compiling Sierra to Casm".to_string()
+    } else {
+        format!(
+            "Compiling Sierra to Casm ({})",
+            sierra_file_path.canonicalize_utf8()?
+        )
+    };
+    spinner.set_message(message);
+
+    Ok(spinner)
+}
+
+fn is_temp_file(file_path: &Utf8PathBuf) -> Result<bool> {
+    let temp_dir = env::temp_dir().canonicalize()?;
+    Ok(file_path.canonicalize()?.starts_with(temp_dir))
+}
```

Comment:
I wonder if there are cases in which this will not correctly recognise the temp dir. 
Though since it's just for outputs I guess it's not a problem anyway.

Reply:
It'd be problematic if we want to test it, but I haven't found any simple way to test it anyway. Other than that as you said it is just for output.